### ISSUE TITLE
[a11y]: FileUploaderItem creates invalid 'aria-describedby' property

### ISSUE THREAD

**Issue by gmoehler** [carbon-design-system/carbon#16524]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.55.0

### React version

16.14

### Automated testing tool and ruleset

IBM Accessibility Checker

### Description

There are 2 cases in which an invalid 'aria-describedby' property is created:

#### Case 1: When `invalid` is true, but `errorSubject` is not provided
In this case. no error element is created, but there still exists an 'aria-describedby' property which now points nowhere. Solution:
FileUploaderItem.tsx [l. 146] 
```
aria-describedby={invalid && errorSubject && `${name}-id-error`}
```

#### Case 2: When a filename with spaces is provided
Here the 'aria-describedby' property will contain spaces which is not allowed. 
Solution: filter out spaces (and other no-valid characters) from the name property before setting the 'aria-describedby' and `id` properties

### WCAG 2.1 Violation

The 'id' "\<name\>-id-error" specified for the ARIA property 'aria-describedby' value is not valid.
[IBM 4.1.2 Name, Role, Value](https://www.ibm.com/able/requirements/requirements/#4_1_2)

#### Case 1:
```
<FileUploaderItem  name="filename" invalid={true} />
```

#### Case  2:
```
<FileUploaderItem  name="filen ame" />
```

#### Case  1:
Set `invalid` but no `errorSubject` and run the Accessibility Checker. You will receive above violation.

#### Case  2: 
Set `name` to a filename containing spaces and run the Accessibility Checker. You will receive above violation.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
